dask/dataframe/io.py

```diff
@@ -462,14 +463,18 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
                  "must preserve the order of its input")
 
     # handle lock default based on wether we're writing to a single entity
+    _actual_get = get or _globals.get('get') or get_sync
```

Comment:
OK, so thinking about this more.  We probably do want to use `DataFrame._get` as the default *if* we're writing to many files/groups.  We only want to use the single-threaded scheduler by default if we have to write things in sequence.  The logic here gets a bit trickier but adding this logic will, I think, improve performance.